nextcord/automod.py

```diff
@@ -0,0 +1,361 @@
+"""
+The MIT License (MIT)
+
+Copyright (c) 2022-present tag-epic
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, overload
+
+from .enums import ActionType, EventType, KeywordPresetType, TriggerType, try_enum
+from .mixins import Hashable
+from .utils import MISSING, snowflake_time
+
+if TYPE_CHECKING:
+    from datetime import datetime
+    from typing import List, Optional, Union
+
+    from .abc import GuildChannel
+    from .errors import InvalidArgument
+    from .guild import Guild
+    from .member import Member
+    from .role import Role
+    from .state import ConnectionState
+    from .types.automod import (
+        AutoModerationAction as AutoModerationActionPayload,
+        AutoModerationRule as AutoModerationRulePayload,
+        TriggerMetadata as TriggerMetadataPayload,
+    )
+    from .types.snowflake import Snowflake
+
+
+__all__ = ("AutoModerationRule", "AutoModerationAction")
+
+
+class AutoModerationAction:
+    """Represents an auto moderation action.
+
+    Attributes
+    -----------
+    type: :class:`ActionType`
+        The action type of this action.
+    notify_channel_id: Optional[:class:`int`]
+        The ID of the channel that this action will notify in.
+    timeout_seconds: Optional[:class:`int`]
+        The number of seconds this rule should timeout when someone triggered the rule.
+    """
+
+    def __init__(self, data: AutoModerationActionPayload):
+        self.type: ActionType = try_enum(ActionType, data["type"])
+        metadata: dict[str, Union[Snowflake, int]] = data.get("metadata", {})  # type: ignore
+        self.notify_channel_id: Optional[int] = (
+            int(metadata.get("channel_id"))  # type: ignore
+            if metadata.get("channel_id") is not None
+            else None
+        )
+        self.timeout_seconds: Optional[int] = metadata.get("duration_seconds")  # type: ignore
```

Comment:
```suggestion
        metadata = data.get("metadata", {})
        self.notify_channel_id: Optional[int] = (
            int(metadata.get("channel_id"))  # type: ignore
            if metadata.get("channel_id") is not None
            else None
        )
        self.timeout_seconds: Optional[int] = metadata.get("duration_seconds")
```
the type checker knows what `data["metadata"]` is as its a typeddict, so therefore it knows `duration_seconds` is an `int` if included

Reply:
sounds unbelieveable but yeah

Reply:
You have all the tools at your disposal to test this locally btw